components/Modal/Modal.tsx

```diff
@@ -45,6 +45,21 @@ export const Modal: React.FC<IModalComponentProps> = withMooskinContext((props)
 		}
 
 		return React.Children.map(children, (child, i) => {
+			// on press esc button close modal
+			const escButtonHandler = (event:any) => {				
```

Comment:
This should be at the root of the component, no need to initialize the function for each child, this will have performance issues.